packages/core/src/RadioCheckToggleCard/RadioCheckToggleCard.tsx

```diff
@@ -0,0 +1,216 @@
+`
+
+const CardHeader = styled(Flex)`
+  min-width: 0;
+  width: 100%;
+
+  & > svg {
+    border: 1px solid var(--icon-border-color);
+    border-radius: var(--icon-border-radius);
+    color: var(--icon-color);
+    padding: 2px;
+    box-shadow: inset 0 0 0 5px var(--icon-border-shadow-color);
+  }
+
+  & + * {
+    margin-top: ${themeGet('space.3')};
+  }
+`
+
+const CardContent = styled(Box)`
+  flex: 1;
+
+  & > :first-child + * {
+    margin-top: ${themeGet('space.3')};
+  }
+`
+
+const RadioIcon = ({ checked, ...props }) => {
+  return checked ? <RadioChecked {...props} /> : <RadioEmpty {...props} />
+}
+
+const CheckboxIcon = ({ checked, ...props }) => {
+  return checked ? <BoxChecked {...props} /> : <BoxEmpty {...props} />
+}
+
+const ToggleIcon = ({ checked, onChange, ...props }) => {
+  const checkIcon = <Check size='16' />
+  return (
+    <Box {...props}>
+      <Toggle height={24} isOn={checked} icon={checkIcon} onToggle={onChange} />
+    </Box>
+  )
+}
+
+const buttonIcon = (cardType: TCardTypes) => {
+  if (cardType === 'radio') {
+    return RadioIcon
+  } else if (cardType === 'checkbox') {
+    return CheckboxIcon
+  } else if (cardType === 'toggle') {
+    return ToggleIcon
+  }
+}
+
+const RadioCheckToggleCard = (props: IRadioCheckToggleCard) => {
+  const {
+    children,
+    cardType,
+    hPosition,
+    vPosition,
+    isHeightOccupied,
+    title,
+    isTitleBold,
+    name,
+    value,
+    onChange,
+    isSelected,
+  } = props
+
+  const id = `${name}_${value}`
+  const DisplayButton = buttonIcon(cardType)
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => onChange(e)
+
+  return (
+    <label htmlFor={id} data-testId='rdt-card-label'>
+      <CombinedInput
+        id={id}
+        name={name}
+        value={value}
+        type={cardType === 'radio' ? 'radio' : 'checkbox'}
+        onChange={handleChange}
+        checked={isSelected}
+      />
+      <RCTCardContainer flexDirection='column' data-testId='rdt-card-main'>
+        <CardHeader
+          alignItems={vPosition}
+          flexDirection={hPosition === 'left' ? 'row-reverse' : 'row'}
+          data-testId='rdt-card-header'
+        >
+          <CardContent>
+            <Text
+              bold={isTitleBold}
+              fontSize={isTitleBold ? '16px' : '14px'}
+              style={{ lineHeight: '24px' }}
+              data-testId='rdt-card-title'
+            >
+              {title}
+            </Text>
+            {isHeightOccupied && children}
+          </CardContent>
+          <DisplayButton
+            checked={isSelected}
+            size='24'
+            onChange={handleChange}
+            style={{
+              ['--icon-border-radius']: cardType === 'checkbox' ? '4px' : '9999px',
+              borderWidth: '1px',
+              margin:
+                hPosition === 'left'
+                  ? `0 var(--default-card-margin) 0 0`
+                  : `0 0 0 var(--default-card-margin)`,
+            }}
```

Comment:
Using `style` and CSS variables is a bit out of pattern for DS; could you achieve the same effect using props and Styled Components?

Reply:
Hey @craigpalermo, I used the CSS variables here because it is better to manage with different items (like background color, icon color, border color) and easy to see what is going to change between different states (like hover, focus and active) rather than keep setting the same CSS properties.  

If it is a bit out of pattern for DS. I am ok to change another pattern for them.